### PR TITLE
Add some missing types

### LINKED ISSUE #27: Add missing TS types
There are still some missing types in `index.d.ts` that cause TypeScript errors:
1. `renderItem` function can return `undefined` – it's useful for filtering options
2. `ItemSeparatorComponent` is completely missing, it can be passed as a prop to change options separator or (when `false`) to remove it completely.

### PR DESCRIPTION
**Pull request by drastus**
Fixes #27

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ interface AutocompleteDropdownProps {
   textInputProps?: object
   onChangeText?(text: string): any
   onSelectItem?(item: TAutocompleteDropdownItem): any
-  renderItem?: (item: TAutocompleteDropdownItem, searchText: string) => JSX.Element
+  renderItem?: (item: TAutocompleteDropdownItem, searchText: string) => JSX.Element | undefined
```
Comment (onmotion) [inline src/index.d.ts:33]: why not just return empty jsx`<></>`?

File: src/index.d.ts [hunk 2]
```diff
@@ -45,6 +45,7 @@ interface AutocompleteDropdownProps {
   ClearIconComponent?: JSX.Element
   ScrollViewComponent?: JSX.Element
   InputComponent?: JSX.Element
+  ItemSeparatorComponent?: JSX.Element | false
```
Comment (onmotion) [inline src/index.d.ts:48]: why do you need false? I think `ItemSeparatorComponent?: JSX.Element` would be enough

**Review by onmotion — CHANGES_REQUESTED** [review]

File: src/index.d.ts (hunk 1 above)
Comment (drastus) [reply src/index.d.ts]: Indeed, it works. Removed `undefined`.

File: src/index.d.ts (hunk 2 above)
Comment (drastus) [reply src/index.d.ts]: You're right, removed `false`.

**Pull request merged**